src/mscorlib/corefx/System/Globalization/SortVersion.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Contracts;
+
+namespace System.Globalization 
+{
+    [Serializable]
+    public sealed class SortVersion : IEquatable<SortVersion>
+    {
+        private int _nlsVersion;
+        private Guid _sortId;
+
+        public int FullVersion 
+        {
+            get 
+            {
+                return _nlsVersion;
+            }
+        }
+
+        public Guid SortId 
+        {
+            get 
+            {
+                return _sortId;
+            }
+        }
+
+        public SortVersion(int fullVersion, Guid sortId) 
+        {           
+            _sortId = sortId;
+            _nlsVersion = fullVersion;
+        }
+
+        internal SortVersion(int nlsVersion, int effectiveId, Guid customVersion) 
+        {
+            _nlsVersion = nlsVersion;
+
+            if (customVersion == Guid.Empty) 
+            {
+                byte[] b = BitConverter.GetBytes(effectiveId);
+                byte b1 = (byte) ((uint) effectiveId >> 24);
```

Comment:
@jkotas the SortVersion code mainly picked up from the desktop. I can try to fix it accordign to your comments